archethic-foundation/archethic-node
# [Feature] Add node shared secrets chain explorer#411

# Description

It adds a new feature Node shared secrets chain explorer to UI, which includes  :node_shared_secrets  Tranasction type in chronological order with number of nodes authorised with arrival of that Transaction.

- An unauthorised node will display a empty UI.
  - An unauthorised node must wait to display NSS chain explorer.
  - `/explorer/chain/node_shared_secrets` route with parameters(page), will still display an empty UI.
- Though a authorised node will display  
   - Total number of authorised nodes in banner
   - Number of nodes authorised w.r.t arrival of that Transaction. 

Fixes # (#411)
- [ ] #411

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
  - A temporary fix to Failure of faucet_Controller_test.exs using Process.sleep
  
- New feature (non-breaking change which adds functionality)
  - Node shared secrets chain explorer.
  
- This change requires a documentation update
  - Brief Info was added to blue banner , describing Node Shared Secrets Chain.

# How Has This Been Tested?
   - Change in parameters > nb_pages < 1, renders page 1.
   - switch page (on page 1) when transaction reaches page limit

- [x] Distributed Testing

    - Authorised nodes display NSS chain
    - The un-authorised nodes should not display NSS Chain.
    
![Screenshot_20220921_004940](https://user-images.githubusercontent.com/90304197/191346042-51485dca-1c8a-47cc-818d-a679711a149f.png)

![Screenshot_20220921_004929](https://user-images.githubusercontent.com/90304197/191346879-0f60aa29-aea8-48d7-9c4d-92bfafb51a6b.png)

    - Page parameters does not break the view, when Node not authorised.
    
![Screenshot_20220921_005003](https://user-images.githubusercontent.com/90304197/191346308-6f073736-11c0-4119-b3af-b80adf60a5f0.png)


    - Display authorised nb_of_nodes
    
![Screenshot_20220921_005045](https://user-images.githubusercontent.com/90304197/191346981-456a5208-d2d5-4839-8f40-8a5dd51af7c2.png)


# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules
